tests/modelling/test_engine.py

```diff
@@ -430,3 +430,43 @@ def test_drawdown(self):
                 columns=assets,
             ),
         )
+
+
+class MultiColumnLoaderTestCase(TestCase):
+    def setUp(self):
+        self.assets = [1, 2, 3]
+        self.dates = date_range('2014-01-01', '2014-02-01', freq='D', tz='UTC')
+
+        asset_info = make_simple_asset_info(
+            self.assets,
+            start_date=self.dates[0],
+            end_date=self.dates[-1],
+        )
+        self.asset_finder = AssetFinder(asset_info)
+
+    def test_engine_with_multicolumn_loader(self):
+        open_, close = USEquityPricing.open, USEquityPricing.close
+
+        loader = MultiColumnLoader({
+            open_: ConstantLoader(dates=self.dates,
+                                  assets=self.assets,
+                                  constants={open_: 1}),
+            close: ConstantLoader(dates=self.dates,
+                                  assets=self.assets,
+                                  constants={close: 2})
+        })
+
+        engine = SimpleFFCEngine(loader, self.dates, self.asset_finder)
+
+        factor = RollingSumDifference()
+
+        result = engine.factor_matrix({'f': factor},
+                                      self.dates[2],
+                                      self.dates[-1])
+        self.assertIsNotNone(result)
+        self.assertIn('f', result)
+        self.assertFalse(result['f'].empty)
+        # (close - open) * window = (1 - 2) * 3 = -3
+        # skipped 2 from the start, so that the window is full
+        self.assertEqual(list(result['f']),
```

Comment:
Rather than casting both of these to lists, can we use `zipline.utils.test_utils.check_arrays` to actually check the numpy arrays here?  I've had bugs in this code where I was expecting an array in one place and returning a Series or vice versa, and just casting both inputs to lists will mask that error.